### PR TITLE
ABC and other changes

### PR DESCRIPTION
**Pull request by MarkusRabe**
- Included circuit simplification with ABC
- Adopted your travis_install.py script for installation
- removed remnants of our wheel experiments

### CODE REVIEW

**Comment by MarkusRabe** [comment]
Hey, I now also unified how ABC, AIGER and CADET are being installed. 

There just one weird thing going on: on my machine, suddenly one testcase in test_count.py is failing. But it works fine on travis. I even added the failing input as an explicit test (to circumvent the randomness in hypothesis), but it still does not fail. Let's check it out when you're back.

File: aiger_analysis/test_count.py [hunk 1]
```diff
@@ -46,3 +46,9 @@ def test_count_and(circ1, circ2):
 def test_count_le(i):
     expr = atom(4, 'x') < atom(4, i)
     assert count.count(expr.aigbv.aig) == i + 8
+
+
+def test_count_le_zero():
```
Comment (mvcisback) [inline aiger_analysis/test_count.py]: you can also use the hypothesis decorator `@example(0)` to force a particular example.

File: README.md [hunk 2]
```diff
@@ -8,3 +8,5 @@
 
 
 Please look up the documentation of <a href="https://github.com/mvcisback/py-aiger">py-aiger</a>.
+
+This package currently assumes `cadet`, `abc`, and `aigtoaig` installed in the PATH.
```
Comment (mvcisback) [inline README.md]: Maybe link to the tool pages?

File: aiger_analysis/cadet.py [hunk 3]
```diff
@@ -57,6 +59,7 @@ def simplify_quantifier_prefix(quantifiers):
     seen_variables = set()
     simplified_q = []
     for q, variables in quantifiers:
+        variables = list(map(str, variables))
```
Comment (mvcisback) [inline aiger_analysis/cadet.py]: Why not just `set(map(str, variables))`.

File: aiger_analysis/cadet.py [hunk 4]
```diff
@@ -57,6 +59,7 @@ def simplify_quantifier_prefix(quantifiers):
...
         assert len(seen_variables & set(variables)) is 0
         seen_variables |= set(variables)
```
Comment (mvcisback) [inline aiger_analysis/cadet.py]: Can drop `set(.)` if map casts variables to a set from the get-go

File: aiger_analysis/cadet.py [hunk 5]
```diff
@@ -82,9 +85,11 @@ def is_true_QBF(e, quantifiers):
             assert quantifiers[0][0] is 'e'
             return is_satisfiable(aig)
     elif len(quantifiers) is 2:  # 2QBF
-        if quantifiers[0][0] is 'e':
+        true_return_code = 10
```
Comment (mvcisback) [inline aiger_analysis/cadet.py]: might be easier to read if we make an enum for the error codes:

```python
from enum import Enum

class CadetCodes(Enum):
     MEANINGFUL_NAME = 10
     ALSO_MEANINGFUL_NAME = 2
```

And then `true_return_code = CadetCodes.Meaningful_Name`

File: setup.py [hunk 6]
```diff
@@ -1,23 +1,22 @@
...
 
-class MyBuild(distutils.command.build.build):
-    '''
-    Custom build class to enable compilation of CADET.
-    Adapted from python-sat.
-    '''
-    def run(self):
-        # download and compile cadet
-        install_cadet.install()
+# class MyBuild(distutils.command.build.build):
```
[the remark this answers is not in the document]
Comment (MarkusRabe) [reply setup.py]: Yeah. We're probably not going to need it anytime soon.

File: README.md (hunk 2 above)
Comment (MarkusRabe) [reply README.md]: Good point.

File: aiger_analysis/cadet.py (hunk 3 above)
Comment (MarkusRabe) [reply aiger_analysis/cadet.py]: Played around with it. In the end it should better be a list and preserve the order of variables, I feel.

File: aiger_analysis/cadet.py (hunk 5 above)
Comment (MarkusRabe) [reply aiger_analysis/cadet.py]: Yeah, probably good for documentation purposes. But makes the code more cluttered.

**Comment by mvcisback** [comment]
Looks good

**Pull request merged**